File: android/app/src/main/java/com/giang10a1dz/googleplayclone/open3rd/Open3rdModule.java
package com.giang10a1dz.googleplayclone.open3rd;

import android.content.Intent;
import android.util.Log;

import androidx.annotation.NonNull;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

public class Open3rdModule extends ReactContextBaseJavaModule {
    Open3rdModule(ReactApplicationContext context) {
        super(context);
    }

    @NonNull
    @Override
    public String getName() {
        return "Open3rdModule";
    }

    @ReactMethod
    public void openApp(String name) {
        Log.d("CalendarModule", "Create event called with name: 11111111111111"+ name);
        Intent launchIntent = getReactApplicationContext().getPackageManager().getLaunchIntentForPackage(name);
        Log.d("CalendarModule", "Create event called with name: 11111111111111" + launchIntent);
        if (launchIntent != null) {
            getReactApplicationContext().startActivity(launchIntent);
        }
    }
}

File: android/app/src/main/java/com/giang10a1dz/googleplayclone/Open3rd/Open3rdModule.kt
package com.giang10a1dz.googleplayclone.open3rd;

import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod

class CalendarModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
    override fun getName() = "CalendarModule"

    @ReactMethod
    fun createCalendarEvent() {
        Log.d("CalendarModule", "Create event called with name")
        Intent launchIntent = getPackageManager().getLaunchIntentForPackage("com.package.address");
        if (launchIntent != null) { 
            startActivity(launchIntent);//null pointer check in case package name was not found
        }
    }
}